sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1427,6 +1427,15 @@ class DataFrame private[sql](
     rdd.mapPartitions(f)
   }
 
+  /**
+    * Returns a new RDD by applying a function to each partition of this DataFrame, while tracking the index
+    * of the original partition.
+    * @group rdd
+    */
+  def mapPartitionsWithIndex[R: ClassTag](f: Iterator[Row] => Iterator[R]): RDD[R] = {
```

Comment:
Except for branch-1.6 issue others pointed, doesn't `rdd.mapPartitionsWithIndex` accepts `f: (Int, Iterator[T]) => Iterator[U]`? The give function doesn't have the partition index parameter?

Reply:
Yeah I expected partition index being exposed to DataFrame API, not just calling rdd.mapPartitionsWithIndex which finally ends up with RDD. It's just a shortcut and high chance to be rejected.